internal/error.go

```diff
@@ -406,9 +408,10 @@ func (e *ApplicationError) Error() string {
 	return e.message
 }
 
-// OriginalType returns original error type represented as string.
-func (e *ApplicationError) OriginalType() string {
-	return e.originalType
+// Type returns error type represented as string.
+// This type can be passed explicitly to ApplicationError constructor or will be set automatically for any other go errors.
```

Comment:
Explain how it is set for other go errors. or even better explain in which situation other errors are converted to this.